src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -234,24 +237,59 @@ private void loadSchematic()
 
         try
         {
-            job.setSchematic(new SchematicWrapper(world, workOrder.getSchematicName()));
+            job.setStructure(new StructureWrapper(world, workOrder.getStructureName()));
         }
-        catch (IllegalStateException e)
+        catch (RuntimeException e)
         {
-            Log.getLogger().warn(String.format("Schematic: (%s) does not exist - removing build request", workOrder.getSchematicName()), e);
-            job.setSchematic(null);
+            Log.getLogger().warn(String.format("StructureProxy: (%s) does not exist - removing build request", workOrder.getStructureName()), e);
+            job.setStructure(null);
             return;
         }
 
-        job.getSchematic().rotate(workOrder.getRotation());
-        job.getSchematic().setPosition(pos);
+        Colony colony = worker.getColony();
+        if(workOrder instanceof WorkOrderBuildDecoration)
+        {
+            job.getStructure().rotate(workOrder.getRotation());
+        }
+        else if(colony != null)
+        {
+            if(workOrder.getRotation() == 0)
+            {
+                IBlockState blockState = world.getBlockState(pos);
+                if (blockState.getBlock() instanceof AbstractBlockHut)
+                {
+                    job.getStructure().rotate(getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING)));
+                }
+            }
+            else
+            {
+                job.getStructure().rotate(workOrder.getRotation());
+            }
+        }
+        job.getStructure().setPosition(pos);
         workOrder.setCleared(false);
     }
 
+
+    private int getRotationFromFacing(EnumFacing facing)
+    {
+        switch (facing)
+        {
+            case SOUTH:
+                return 2;
+            case EAST:
+                return 1;
+            case WEST:
+                return 3;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)